ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/LayeredPendingTransactions.java

```diff
@@ -95,66 +95,116 @@ public synchronized TransactionAddedResult addLocalTransaction(
   TransactionAddedResult addTransaction(
       final PendingTransaction pendingTransaction, final Optional<Account> maybeSenderAccount) {
 
-    final long senderNonce = maybeSenderAccount.map(AccountState::getNonce).orElse(0L);
+    final long stateSenderNonce = maybeSenderAccount.map(AccountState::getNonce).orElse(0L);
 
-    logTransactionForReplayAdd(pendingTransaction, senderNonce);
+    logTransactionForReplayAdd(pendingTransaction, stateSenderNonce);
 
-    final long nonceDistance = pendingTransaction.getNonce() - senderNonce;
+    if (nonceSkewDetected(pendingTransaction, stateSenderNonce)) {
+      reconcileSender(pendingTransaction.getSender(), stateSenderNonce);
+    }
+
+    final long nonceDistance = pendingTransaction.getNonce() - stateSenderNonce;
 
     final TransactionAddedResult nonceChecksResult =
-        nonceChecks(pendingTransaction, senderNonce, nonceDistance);
+        nonceChecks(pendingTransaction, stateSenderNonce, nonceDistance);
     if (nonceChecksResult != null) {
       return nonceChecksResult;
     }
 
     try {
-      TransactionAddedResult result =
-          prioritizedTransactions.add(pendingTransaction, (int) nonceDistance);
-
-      if (result.equals(REORG_SENDER)) {
-        result = reorgSenderOf(pendingTransaction, (int) nonceDistance);
-      }
-
-      return result;
+      return prioritizedTransactions.add(pendingTransaction, (int) nonceDistance);
     } catch (final Throwable throwable) {
-      // in case something unexpected happened, log this sender txs and force a reorg of his txs
+      return reconcileAndRetryAdd(
+          pendingTransaction, stateSenderNonce, (int) nonceDistance, throwable);
+    }
+  }
+
+  private TransactionAddedResult reconcileAndRetryAdd(
+      final PendingTransaction pendingTransaction,
+      final long stateSenderNonce,
+      final int nonceDistance,
+      final Throwable throwable) {
+    // in case something unexpected happened, log this sender txs and force a reorg of his txs
+    LOG.warn(
+        "Unexpected error {} when adding transaction {}, current sender status {}",
+        throwable,
+        pendingTransaction.toTraceLog(),
+        prioritizedTransactions.logSender(pendingTransaction.getSender()));
+    LOG.warn("Stack trace", throwable);
```

Comment:
warn seems high for a stack trace

Reply:
this at `WARN` while the new txpool is in preview, since it helps me a lot to understand the source of any unexpected exception, will add a ToDo to demote it when the layered txpool will be enabled by default